### ISSUE TITLE
Recurring: Dependency Scan

### ISSUE THREAD

**Issue by elopatin-uc3** [CDLUC3/mrt-doc#1463]

**Comment by elopatin-uc3** [comment]
Dependency scan Feb 28:

Ruby
- Merritt UI has multiple low severity alerts
- Merritt Docker: two Moderate alerts
- Merritt atom: two Nokogiri low severity alerts
- mrt-cron: similar Nokogiri alerts
- UC3 admin: Nokogiri alerts

Java
- core2: high severity alert for MySQL

Tag reports
- will hold off on these until cleanup tools are in place

Java Version
- waiting on new microservice servers

Maven, Maven plug-ins, docker version
- waiting on new dev boxes

SecUnit
- three items related to mrt-integ-tests. One CVE with fix available: labexpat1 CVE-2024-45491

We will create a tickets to address each of the above noted alert items and the SecUnit item with an available fix.

**Comment by terrywbrady** [comment]
Related to this review 
- #2175

**Comment by elopatin-uc3** [comment]
@marisastrong This ticket has been created: https://github.com/CDLUC3/mrt-doc/issues/2213
@terrywbrady @dloy

**Comment by elopatin-uc3** [comment]
No Java dependency updates noted by Dependabot
Low and moderate updates needed to Ruby implementations

David will scan for BOM dependency updates (ticket forthcoming)

Tomcat: Review post-credential

[Maven plug-ins:](https://github.com/CDLUC3/mrt-doc-private/blob/main/docs/dependency_scans.md#maven-plugins)
[Registry of Plugins Used by Merritt](https://github.com/CDLUC3/mrt-core2/blob/main/parprop/pom.xml)
[docker-maven-plugin](https://github.com/fabric8io/docker-maven-plugin/releases)
[jacoco-maven-plugin](https://github.com/jacoco/jacoco/releases)
[maven-assembly-plugin](https://github.com/apache/maven-assembly-plugin/tags)
[maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin/releases)
[maven-surefire-plugin](https://github.com/apache/maven-surefire/releases)
[maven-failsafe-plugin](https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-failsafe-plugin)
[maven-javadoc-plugin](https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-javadoc-plugin)

Javadoc update and others: ticket forthcoming
https://github.com/CDLUC3/mrt-core2/blob/main/parprop/pom.xml

**Comment by elopatin-uc3** [comment]
Dependency Scan, May 30, 2025:

Java:
- MySQL connector for mrt-core2: still no available patch
- All other Java repos clear

Ruby:
- No alerts!

Repository tags:
- Old Admin tool: no cleanup needed yet
- New Admin tool: no cleanup needed
- UI: Many tags in the past week: could be cleaned up at next scan meeting
- Ingest: To clean up builds that are no longer active
- Store: To clean up builds that are no longer active
- Inventory: Deleted one build from last Oct.
- Audit: To clean up Audit builds from last year
- Replic: Same as Audit

Java version:
- Post-changeToken (which will complete soon)

Tomcat:
- Evaluate Tomcat 10 when updating Java

Code Build
- Operating as expected; no changes needed

Maven Plugins
- ToDo and ticketed; delayed by higher-priority work
- re-evaluate when implementing newer version of Java

Docker
- Updates via recent work on Docker dev boxes

SecUnit:
- Old Docker boxes to be retired soon

**Comment by ashleygould** [comment]
review of trivey scan - lots of vulnerabilities marked as "do-not-fix".  what is our plan for these?

**Comment by terrywbrady** [comment]
By trivy, I presume this is SecUnit.

I do not know why things are do-not-fix.  We had several things with no upstream fix for a long time.

I can look at this when I am back in the office.  It would be helpful to have a ticket created for me with the details.

I am rushing to do something for @elopatin-uc3 before I leave.

**Comment by terrywbrady** [comment]
SecUnit still indicates that these have "No upstream fix".  The "will-not-fix" indicator may be passing along a status from the upstream containers.

I believe we are generally using very standard docker images.  

Perhaps @ashleygould and @terrywbrady  could review the findings together to make sure that we concur.

**Comment by elopatin-uc3** [comment]
Dependency scan - 2025-09-03

Java:
- Still no fix for the MySQL connector CVE
- Otherwise, no other Java dependency issues

Ruby:
- mrt-atom: Moderate Active record logging vulnerability: Ticket forthcoming
- No other Ruby issues

Tag reports (review via New Admin tool - Source > Repository Tags...)
- None flagged by SecUnit

MySQL
- Minor update being pushed out by IAS 
- check in with Joe or Martin re: schedule

Tomcat
- When it becomes available, we've decided to update to 9.1 per our version of Jersey
  - availability TBD
  - eventually we'll want to move to 11, after we move Production to ECS
 
Maven
- Currently on 3.8.4
- 3.9.11 was released in July 2025; we could consider updating to this

Maven plug-ins
- Will determine if we update on next scan

SecUnit
- Will redeploy old Admin to resolve a Nokogiri update

**Comment by elopatin-uc3** [comment]
Dependency Scan - 2025-09-26

#### Java
ZK 3.9.4 update - confirming we're already on 3.9.4
MySQL - still no patch

Merritt ZK lib - no alerts

#### Ruby - UI, cron, old and new Admin
No alerts

#### Available JAR updates
- Will run process to detect these and incorporate into next BOM update
  - BOM 4.0 just released

ZK server and Java versions - Recently updated and reviewed
- We are within our specified review cycle window of 6 mos.)

#### Tomcat
- 9.0.104: to patch to 9.0.108 per Ashley

#### Maven plug-ins
- To check ahead of next dependency scan

#### Ruby version
- Ticketed update to 3.3 or 3.4 (#2326) 

#### SEC Unit
- Built mrt-dashboard image in August had image-magick vulnerability; deleted this image as newer ones contain fix
  - Note: We do not use image-magick; it's just bundled into the base image we use

**Comment by terrywbrady** [comment]
Dependency scan - 2025-10-31

- Java
  - one outstanding mysql vulnerability with no fix
  - 2 new moderate vulnerabilities.... will pick up in the next round of BOM updates
 
- Ruby
  - None

- Zookeeper
  - 3.9.4 is available.  We should consider working on this after the next dependency meeting
 
- Other
  - It is time to evaluate the ruby and rails versions, we have that in queue for an upcoming sprint
  - Tag reports removed from the Dependency Scan script
  - Replaced with a link to the admin tool to do an Image and Artifact review every 2-3 months

**Comment by elopatin-uc3** [comment]
Dependency scan - 2025-12-04

- Java
  - core2: Still no patch version for mysql connector issue
     - Moderate: logback: will update to 1.5.19
  - ZK: Moderate: Insufficient permission check in AdminServer: fix in 3.9.4 (will plan to do in new year)
  - No other updates needed

ZK lib: no updates
Merritt UI: no updates

Merritt docker: no updates
Merritt atom: no updates
Int tests: no updates
Ingest workspace: no updates
New Admin: no updates

Tika: Does not yet need a review
JAR updates: Review in January

Java 25: potential 10-15% JRE performance boost. To review in March 2026

Tomcat: Review 10.1 in January

Maven: Review in March 2026
Maven plug-ins: Review in April 2026

Ruby upgrade to 3.4.3 for Ruby version and Rails 8.1 in progress (on Stage)
- Ruby EOL: 3.4.7 ends in 2 years
  - Will do a minor update from 3.4.3 to 3.4.7 in the near future
- 8.1 EOL: 10 Oct 2027

JQuery eval next year.

SecUnit:
- No issues noted

**Comment by elopatin-uc3** [comment]
#### Dependency Scan, 2025-04-03:

Java
- Clear per David's BOM 6.1 update that was just merged
Ruby/Rails
- Admin clear per recent gem updates taking place
- UI release needed to apply gem updates
- Current Ruby is 3.4
- Current Rails 8.1
ZK
- BOM updated to 3.9.5
- Server on Stage update set for next week
- ECS also needs to be updated
LDAP: Updating schedule to check every 3 months
- Just updated to latest version
Java
- 25? Team decided we're good with 21 (next review six months from now)
Tomcat:
- 9.0.116 is good through 31 March 2027
- Currently on 9.0.115
- David will look to see if/when a 9.1 is available:
  - "Extended support will be provided via a new 9.1.x branch that will be started when 9.0.x reaches end of support."
Maven
- Updated on Docker boxes to 3.9.9 in March
Maven plug-ins
- Set to review in October

SecUnit
- Gem updates made on 4/2 may cancel out some SecUnit alerts on today's report
  - Terry's going to check with Martin; likely we need to wait a day because gem updates were at COB 4/2
  - Aside: is there a way to filter out alerts for other UC3 teams?
- Tags and artifacts reviewed and cleaned-up

**Comment by elopatin-uc3** [comment]
#### 2026-05-01

Java:
mrt-core:
- One high and three moderate;
- Still no fix on the MySQL; will integrate moderate fixes into BOM 7

Ruby code:
- One high sev for Rails 2.7 
- gem updates: create new ticket for ERB version (EL)

Not due to look at ZK or LDAP
No updates needed for Tomcat
Maven and Maven plug-ins OK

Ruby - On 3.4 and we have a good window of time for this; no upgrade needed at this time.

jQuery - Review every 6 months (more frequently than current 1yr interval)
- 3.x is still under support; no end date noted

SecUnit
- Warning re: MinIO; we already have a ticket to replace in dev with RustFS
- Latest versions of Selenium + Chrome and Selenium + Chromium in place, but there is still a low priority issue with these
  - mrt-it-chrome includes the Docker image that uses these Selenium dependencies
- No EC2 findings

Images, artifacts cleaned-up via Admin tool

**Comment by marisastrong** [comment]
Thanks for the review. I can bring up at meeting Monday but see reference to a Ruby 2.7 and a 3.4?  
Also RustFS for MinioIO, not Localstack?

**Comment by elopatin-uc3** [comment]
Thanks for catching that - it should have been Rails 2.7 (edited). 

Here's the RustFS ticket: https://github.com/CDLUC3/mrt-doc/issues/2620

I think Terry learned of another example of a successful MinIO > RustFS implementation. I'm not sure if he's investigated Localstack.

**Comment by elopatin-uc3** [comment]
#### 2026-06-05

Java
- mrt-core 
  - Still no fix for mysql connector
  - log4j update RFC5424Layout log injection (moderate)

ZK (Java and Ruby) - No updates

Ruby
- UI: no updates
- mrt-docker and mrt-atom: no updates
- Admin tool: Faraday (low); will run updates for Admin tool

Tika: Due to check next month
Jars review: Due to check next month
ZK and LDAP: Due to check next month
Java: Check in 4 months
Tomcat 9: Security support until March 2027
- in Docker, tomcat9, jdk21-corretto: latest image for this is being pulled
Maven and Maven plug-ins not due to look at soon
Ruby: to review 2 months 
jQuery: to review 6 month out

SecUnit:
- Much marked as defer to next month
- Today, w/early filter: 201 items
- Some we'll be able to close out because they are in archived versions of artifacts
- `net-imap` in callback, mock-merritt-it and Admin all built on Ruby 3. 
  - If we patch Ruby there are still 5 critical vulnerabilities in latest build of Ruby 3 and 3.4
  - Marked as revisit in 3 weeks
  - Marked archived tags as n/a
- mrt-audit and rustfs, multiple tags/versions: tomcat catalina vulnerabilities
  - CVEs 43512, 43515
  - We're at latest version of tomcat: 9.0.118; these CVEs were addressed in 118; see [ref](https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.116)
  - Marked archived images: these images up to 3.0.2 have the CVEs; however 3.0.3 in production does not
- mrt-dashboard: 
  - some CVEs apply to archived images (marked as such in SU)
  - `net-imap` related CVEs for mrt-dashboard do not yet have a fix
- mrt-ingest 2.4.13 is in production
  - everything earlier than this marked as archived (n/a) images w/CVEs; not deployed
  - mrt-ingest-tests: marked as waiting for updates
- mrt-inventory 2.6.2 is deployed to production
  - everything earlier than this marked as archived (n/a) images w/CVEs; not deployed 
- mrt-replic 3.0.3 is deployed to production
  - everything earlier than this marked as archived (n/a) images w/CVEs; not deployed
  - Replic is not yet in ECS. When we migrate, we'll update to 3.0.4 w/tomcat 9.0.118
- mrt-store 2.6.2 is deployed to production
  - everything earlier than this marked as archived (n/a) images w/CVEs; not deployed
- mysql-ruby-lambda: revisit in three weeks

We did not have time in today's meeting to purge old images.

**Comment by elopatin-uc3** [comment]
#### 2026-08-07

#### Java 
- Two moderate jackson-databind; will address in BOM8 for mrt-core2
#### Ruby apps
- Gem update needed for the UI. We've started a ticket: https://github.com/CDLUC3/mrt-doc-private/issues/307

#### Periodic reviews
Jars, ZK, LDAP will be reviewed in Sept
Java version will be reviewed in December

#### Tomcat
- Just updated to 9.0.119 (EC2) and 9.0.120 (ECS - just pulls in the latest major version)

#### Maven and plugins
- Review in 6 mos.

#### Ruby
- Ruby 3.4.3, good for a year
- Rails 8.1, good for one year and 2 mos.

#### jQuery
- Current version is OK

#### SecUnit
- EC2s
  - Batch servers, ZK boxes, docker boxes have some issues that should be remedied by monthly patching
  - Need to determine when we formally shut down the Docker boxes
  - Batch servers: need to confirm if anything else besides Nuxeo 
  - Clean-up script also runs on Batch server
- EC2 auto scaling group (ASG) boxes
  - Dev, ingest workspace, proxy all have available patches that should be put in place on Monday per weekly restarts 
  - vim-related patches
 
#### ECS
- Patches should be applied at Monday's weekly stack redeploy

General: Need to investigate tagging strategy that may be throwing off Martin's filtering in SecUnit. Will also discuss in next IAS meeting.

#### Build Tags cleaned up via Admin:
- UI: No candidate tags to clean up; all in protected state per recent deployments
- Ingest: 2.4.15, .12, .7, .6 + dev0x tags deleted; deleted 2.4.5 
- Store: 3.0.4 dev0x tags deleted; deploying 3.1.1 to Dev (contains BOM7 work); deleted 2.5.1, .2
- Inventory: deleted 2.5.4
- Audit: deleted 2.5.3 and 2.5.4
- Replic: deleted 2.5.3 and 2.5.4

#### Images associated with branch names:
- UI: deleted uxrev1 and uxrev2
- Ingest: deleted gracefulShutdown, storateManifestOmit, ses 
- Store: deleted v2b7, gracefulShutdown, ignull
- Inventory: n/a
- Audit: n/a
- Replic: deleted v2b7

#### Artifacts associated with branch names:
- Ingest: deleted gracefulShudown snapshot, rustfs snapshot, ses_env, ses_postfix, storageManifestOmit
- Store: 3.0.5.dev01, gracefulShutdown snapshot, v2b7 snapshot
- Inventory: manyfiles snapshot
- Audit: n/a
- Replic: deleted v2b7 snapshot

**Comment by elopatin-uc3** [comment]
#### 2026-09-04

#### Java service dependencies 
- Two new moderate issues; Log4J and JacksonDatabind; will slot into next BOM updates
#### Ruby apps
- No updates needed

#### Periodic reviews
Jar review will be picked up by David in the next few weeks.

#### ZooKeeper
- Currently running the latest released/stable version of ZK: v3.9.5

#### LDAP
- Due for review now
- Currently running the latest released version: OpenDJ 5.1.2

#### Java runtime and compiler versions
- Not due to review this month

#### Tomcat
- ECS pulls in the latest major version offered in tomcat:9-jdk21-corretto
- We should watch for when Corretto moves to 9.1

#### Maven and plugins
- Review in 5 mos.

#### Ruby
- Not due to review this month

#### jQuery
- Not due to review this month

#### SecUnit
EC2:
- Tomcat vulnerability flagged in all of our Docker images
- This is not flagged yet for EC2 services in SecUnit. 
  - A fix is available in 9.0.121 when needed

ECS - migrated services
- Patches applied during weekly/Monday stack redeploy
- Monday's redeploy will clear the Tomcat issue for all ECS-migrated services
